### PR TITLE
Connect to api

### PR DESCRIPTION
**Pull request by mbdesigns1989**
- I have added API calls. 

- The app can fetch data from the API.

- The app can add new data to the API.

- The app can remove data from the API.

### CODE REVIEW

File: src/components/AddBook.js [hunk 1]
```diff
@@ -1,41 +1,39 @@
...
-    setBook({ title: '', author: '' });
+    await dispatch(addBook(newBook));
+
+    setTitle('');
+    setCategory('');
 
-    titleRef.current.value = '';
-    authorRef.current.value = '';
+    await dispatch(getBooks());
+    document.location.reload(true);
```
Comment (ArthurGC) [inline src/components/AddBook.js:28]: - [OPTIONAL] There is an easier way to add books to your project. Only using `addBook` and `createBook` without reloading the page.
- First, you could remove these lines `await dispatch(getBooks());` `document.location.reload(true);`

File: src/redux/books/books.js [hunk 2]
```diff
@@ -1,20 +1,27 @@
...
+
 const reducer = (state = initialState, action) => {
   switch (action.type) {
     case ADD_BOOK:
-      return [...state, action.payload];
-    case REMOVE_BOOK:
-      return state.filter((book) => book.id !== action.payload.id);
+      createBook(action.payload);
+      return state;
+    case REMOVE_BOOK: {
```
Comment (ArthurGC) [inline src/redux/books/books.js:15]: - [OPTIONAL] Here, It will be `return [...state, action.payload];`

File: src/utils/api.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  return dispatch({ type: GET_BOOKS_SUCCESS, books });
+};
+
+const createBook = (book) => {
+  fetch(booksBaseURI, {
+    method: 'POST',
+    body: JSON.stringify({
+      item_id: book.id,
+      title: book.title,
```
Comment (ArthurGC) [inline src/utils/api.js:21]: - [OPTIONAL] Finally, the id will be `item_id: book.item_id`. 😉 So with that you won't need to reload the page and call your API again.

File: src/components/AddBook.js [hunk 4]
```diff
@@ -1,41 +1,39 @@
...
-    author: '',
-  });
-
-  const submitBookToStore = (e) => {
+  const submitBookToStore = async (e) => {
     e.preventDefault();
 
     const newBook = {
       id: uuidv4(),
-      ...book,
```
Comment (ArthurGC) [inline src/components/AddBook.js:22]: - [OPTIONAL] It will be `item_id: uuidv4(),` ♻️

**Review by ArthurGC — APPROVED** [review]
# Approved project!  💯 ✅ 🥳 🎈 🎉
Hi  @mbdesigns1989 

Amazing job!
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by mbdesigns1989** [comment]
Thanks a lot, Arthur, I will take your suggestions into consideration.

**Pull request merged**